sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1309,7 +1309,8 @@ void ProgramManager::bringSYCLDeviceImagesToState(
         break;
       }
       case bundle_state::executable:
-        // Device image is already in the desired state.
+        DevImage = build(DevImage, getSyclObjImpl(DevImage)->get_devices(),
```

Comment:
Nit: I assume this build call is optionally needed to do native device code linking? Why not call to `link` then? Please add a comment.

Reply:
Build is needed here to create device image which contain spec constants; as device image is in executable state because of AOT, build instead of link (object state) is used.